programs/keosd/main.cpp

```diff
@@ -43,12 +60,23 @@ int main(int argc, char** argv)
          .default_unix_socket_path = keosd::config::key_store_executable_name + ".sock",
          .default_http_port = 0
       });
+      const int parent_id = get_parent_pid(argc, argv);
+      const int argc_no_parent_id = (parent_id > 0) ? argc - 2 : argc;
+      char** const argv_no_parent_id = (parent_id > 0) ? &argv[2] : argv;
+
       app().register_plugin<wallet_api_plugin>();
-      if(!app().initialize<wallet_plugin, wallet_api_plugin, http_plugin>(argc, argv))
+      if(!app().initialize<wallet_plugin, wallet_api_plugin, http_plugin>(argc_no_parent_id, argv_no_parent_id))
          return -1;
       auto& http = app().get_plugin<http_plugin>();
       http.add_handler("/v1/" + keosd::config::key_store_executable_name + "/stop", [&a=app()](string, string, url_response_callback cb) { cb(200, fc::variant(fc::variant_object())); a.quit(); } );
       app().startup();
+
+      if (parent_id > 0)
+      {
+         const int ret_sig_que = kill(parent_id, SIGUSR1);
```

Comment:
What happens if the parent has already exited?
What happens if keosd initialization fails (e.g. someone else launched keosd concurrently).